py3status/modules/thunderbird_calendar.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""
+Display tasks in thunderbird calendar.
+
+Configuration parameters:
+    - cache_timeout : how often we refresh usage in seconds (default: 120s)
+    - profile_path : path to the user thunderbird profile (not optional)
+    - format : see placeholders below
+
+Format of status string placeholders:
+    {due} : due tasks
+    {completed} : completed tasks
+    {current} : title of current running task (sorted by priority and stamp)
+
+Make sure to configure profile_path in your i3status config using the full
+path or this module will not be able to retrieve any information from your
+calendar.
+
+ex: profile_path = "/home/user/.thunderbird/1yawevtp.default"
+
+@author mrt-prodz
+"""
+from sqlite3 import connect
+from os import access, R_OK
+from time import time
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 120
+    # user must configure the path to thunderbird profile_path
+    # ex: /home/user/.thunderbird/1yawevtp.default
+    profile_path = ''
+    format = 'tasks:[{due}] current:{current}'
+
+    # return error occurs
+    def _error_response(self, color):
+        response = {
+            'cached_until': time() + self.cache_timeout,
+            'full_text': 'Err',
+            'color': color
+        }
+        return response
+
+    # return calendar data
+    def get_calendar(self, i3s_output_list, i3s_config):
+        db = self.profile_path + '/calendar-data/local.sqlite'
+        due = completed = 0
+        current = ''
+        if not access(db, R_OK):
+            return self._error_response(i3s_config['color_bad'])
```

Comment:
Maybe you could pass to the user a nice error eg `No access - check profile_path`